src/esp/agent/Agent.cpp

```diff
@@ -31,10 +32,21 @@ Agent::Agent(scene::SceneNode& agentNode, const AgentConfiguration& cfg)
     // sensor
 
     auto& sensorNode = agentNode.createChild();
-    sensors_.add(
-        sensor::PinholeCamera::create(sensorNode, spec));  // transformed within
+    switch (spec->sensorSubtype) {
+      case sensor::SensorSubtype::PINHOLE: {
```

Comment:
SensorSubType::Pinhole
Please use "Pascal" style for the enum.

Reply:
It occured me to we also need a str to enum mapping potentially or allow the ENUMs to be exposed on the habitat_lab side in the YAML configs.

Reply:
getattr should work for this